Add support for any language:

dmitry-timofeev opened the pull request:
Auto-detect the source language and the alias to use using the Pygments API.

dmitry-timofeev commented:
It appears [`codehilite`](https://squidfunk.github.io/mkdocs-material/extensions/codehilite/) is used widely as an markdown extension, and it does not work with all _file_ extensions (e.g., "cs" does not yield good enough results for C# — must be "csharp").

I'll see if the correct info can be extracted using the Pygments API: https://pygments.org/

rnorth commented:
Thanks for looking into this - it's a good idea. I'd be interested to see if there's a suitable source for the extension->language mapping.

dmitry-timofeev commented:
It seems Pygments provides such a mapping through a pair of [`get_lexer_for_filename`](https://pygments.org/docs/api/#pygments.lexers.get_lexer_for_filename) and `aliases`. 
As Pygments is used by [codehilite](https://squidfunk.github.io/mkdocs-material/extensions/codehilite/), it must be the best source of such mapping. I updated the code to use it and added more tests.

rnorth commented on tests/codeinclude/test_languages.py:
```diff
@@ -0,0 +1,18 @@
+        self.assertEquals('rust', get_lang_class('HelloWorld.rs'))
+        self.assertEquals('docker', get_lang_class('Dockerfile'))
+        self.assertEquals('xml', get_lang_class('HelloWorld.xml'))
+        self.assertEquals('toml', get_lang_class('HelloWorld.toml'))
```
Hmm, the test seems to fail on this line - I wonder if there's varying support for the `toml` extension in different versions of pygments.

```
>       self.assertEquals('toml', get_lang_class('HelloWorld.toml'))
E       AssertionError: 'toml' != 'none'
E       - toml
E       + none
```

dmitry-timofeev replied:
It seems I used `Pygments~=2.5` or newer, which version fails for you? Would it make sense to require a certain version of it, or just remove this test?

rnorth commented:
Looks like this is included in #15 
Trying to unpick these and forming a larger PR, #22, where I'll resolve the conflicts that have arisen
